specs/virtual_state/end_tx.md

```diff
@@ -2,67 +2,51 @@
 
 ## Procedure
 
-The `end_tx` gadget performs preliminary checks on a transaction and its caller's account as well as further checks depending on the transaction type. 
+The `end_tx` gadget handles refund and coinbase tip as well as cumulative gas within a block. 
 
-More particularly, for all transactions, we make sure that the account nonce is incremented by 1 if the tx is valid, and that the caller address is not null. We also verify that a transaction is set as invalid if and only if there is either not enough gas, the tx's nonce is not the same as caller's account previous nonce or the balance is insufficient.
```

Comment:
TBH, I like this paragraph (L7 - L13). It explain this gedget in in high level. But it's just my personal preference.

Reply:
I copy pasted being_tx as end_tx and then updated each section but forgot to update this one, hence the text is incorrect here.

That's why I deleted this paragraph (because this is from begin_tx!) and replaced it with:
`The 'end_tx' gadget handles refund and coinbase tip as well as cumulative gas within a block. 
`

Reply:
I wish I could write more but, except for the transition, that's pretty much what the gadget does